lib/service/service.go

```diff
@@ -1084,10 +1084,17 @@ func initUploadHandler(ctx context.Context, auditConfig types.ClusterAuditConfig
 		}
 		return handler, nil
 	case teleport.SchemeS3:
-		config := s3sessions.Config{}
+		var fips *bool
+		if useFIPS {
+			fips = new(bool)
+			*fips = true
+		}
+		config := s3sessions.Config{UseFIPSEndpoint: fips}
```

Comment:
Yeah these are interesting questions.

It might be nice if our FIPS builds that are configured for S3/Dynamo automatically use FIPS endpoints, even if users don't set the URL parameters.

Would need to do some research to determine if this would break anybody.

Reply:
@r0mant and I discussed these points generally on https://github.com/gravitational/teleport/issues/11291, specifically https://github.com/gravitational/teleport/issues/11291#issuecomment-1149165730 

My initial testing and other testing so far shows that the behavior is transparent to the end user regardless of which endpoint is being used, and I haven't seen any breakage yet; I just verified which endpoint was being used and captured the DNS data with Wireshark to prove it. I'll do some more testing tonight and post the results here and on the ticket when I have them.

Reply:
Yeah we discussed this before and had the same concerns which is why we decided to go for the following:

- In Teleport 10 we will use FIPS endpoints by default when you run in `--fips` mode and document this as potentially breaking change (though it shouldn't really break anything).
- In Teleport 9 we won't use FIPS endpoints in `--fips` mode by default, users will have to opt-in.